detekt-gradle-plugin/build.gradle.kts

```
@@ -140,6 +140,7 @@ dependencies {
         }
     }
 
+    compileOnly("io.gitlab.arturbosch.detekt:detekt-api:1.23.8")
     // We use this published version of the detekt-formatting to self analyse this project.
```

nit: Can you move this comment above and make it generic for both dependencies?


```suggestion
    // We use those published version of detekt artifacts to self analyse this project and produce a Problem API report
```